### ISSUE TITLE
buildconfig deletion leftover issue

### ISSUE THREAD

**Issue by odra** [openshift/origin#19862]
Deleting all buildconfigs from a project (`oc delete bc -n myproject --all`) doesn't delete all of them leaving one "leftover" buildconfig.

The leftover buildconfig removes all existing labels and contains the following annotations:

```
jenkins.openshift.io/generated-by: jenkins
jenkins.openshift.io/job-path: myproject/myproject-helloworld-android-pipeline-2
```

##### Version
oc v3.7.2+282e43f
kubernetes v1.7.6+a08f5eeb62
features: Basic-Auth GSSAPI Kerberos SPNEGO

openshift v3.9.0+4814d4a-3
kubernetes v1.9.1+a0ce1bc657

##### Steps To Reproduce

BuildConfig file:

```
kind: BuildConfig
apiVersion: v1
metadata:
  name: helloworld-android-pipeline-2
spec:
  source:
    git:
      uri: https://github.com/odra/helloworld-android-gradle.git
      ref: jenkins-apk-path
  strategy:
    jenkinsPipelineStrategy:
      env:
        -
          name: FH_CONFIG_CONTENT
          value: empty_content
        -
          name: BUILD_CONFIG
          value: debug
      jenkinsfilePath: Jenkinsfile
```

1. Create more than one buildconfig by running `oc create -f  build.yml -n myproject`
2. Delete all buildconfigs `oc delete bc -n myproject --all`

##### Current Result

One buildconfig was not deleted.

##### Expected Result

All buildconfigs should be deleted.

**Comment by php-coder** [comment]
Sounds like all build configs were deleted but one of them was re-created.

CC @openshift/sig-developer-experience

**Comment by bparees** [comment]
@gabemontero @coreydaley I assume this is the known issue w/ the sync plugin recreating the buildconfig after it gets deleted.  Not sure what version of the sync plugin that was fixed in?

**Comment by gabemontero** [comment]
most likely @bparees - assuming that being at openshift v3.9 means they are using the 3.9 version of the jenkins image ... that correlates to v1.0.7, which has the bug you mentioned.

@odra - can you either inspect the jenkins image stream you are using to confirm the version of the image, or log into a jenkins instance that is started when you create the pipeline strategy build config you noted, and confirm the version of the openshif-sync plugin?

The fix got dropped about a week ago into v1.0.19 of the sync plugin, which is only in our pre release 3.11 / latest centos image (that is right, given the source of the change, it didn't make the cut off for 3.10, I did not think it had enough soak time).

**Comment by gabemontero** [comment]
Closing under the assumption this is the known issue noted above.

@odra - please reopen if you still see this after using v1.0.19 or later of the sync plugin.
